pkg/controller/sidecarset/sidecarset_processor.go

```diff
@@ -570,6 +570,8 @@ func updatePodSidecarContainer(control sidecarcontrol.SidecarControl, pod *corev
 			pod.Annotations[sidecarcontrol.GetPodSidecarSetVersionAltAnnotation(olderSidecar)] = sidecarSet.ResourceVersion
 		}
 	}
+	// update sidecarSet hash in pod annotations[kruise.io/sidecarset-hash]
```

Comment:
plz comment in the code that why sidecarset hash should not be patched in UpdatePodAnnotationsInUpgrade